ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java

```diff
@@ -172,12 +217,61 @@ private static TabularRow fromGenericRow(
     return TabularRow.createRow(80, contents, false, 0);
   }
 
-  public static void assertStream(final AssertStream assertStatement) {
-    throw new UnsupportedOperationException();
+  public static void assertStream(
+      final KsqlExecutionContext engine,
+      final AssertStream assertStatement
+  ) {
+    assertSourceMatch(engine, assertStatement.getStatement());
+  }
+
+  public static void assertTable(
+      final KsqlExecutionContext engine,
+      final AssertTable assertStatement
+  ) {
+    assertSourceMatch(engine, assertStatement.getStatement());
   }
 
-  public static void assertTable(final AssertTable assertStatement) {
-    throw new UnsupportedOperationException();
+
+  private static void assertSourceMatch(
+      final KsqlExecutionContext engine,
+      final CreateSource statement
+  ) {
+    final SourceName source = statement.getName();
+    final DataSource dataSource = engine.getMetaStore().getSource(source);
+
+    MUST_MATCH.forEach(prop -> prop.compare(dataSource, statement));
+  }
+
+  private static final class SourceProperty {
+    final Function<DataSource, Object> extractSource;
+    final Function<CreateSource, Object> extractStatement;
+    final String propertyName;
+
+    private SourceProperty(
+        final Function<DataSource, Object> extractSource,
+        final Function<CreateSource, Object> extractStatement,
+        final String propertyName
+    ) {
+      this.extractSource = extractSource;
+      this.extractStatement = extractStatement;
+      this.propertyName = propertyName;
+    }
+
+    private void compare(final DataSource dataSource, final CreateSource statement) {
+      final Object expected = extractStatement.apply(statement);
+      final Object actual = extractSource.apply(dataSource);
+      if (!actual.equals(expected)) {
+        throw new KsqlException(
```

Comment:
Again with the `KsqlException` ;).   How about `AssertError`?

Reply:
`AssertionError` is not part of `Exception` which makes it a little bit of a pain to deal with - though I can change it if you feel super strongly about it.

Reply:
But this is an assertion failure, so using an `AssertError` is the right approach IMHO.  This is a definite 'the test has failed', rather than something else went wrong.  Makes it almost impossible for someone to come along and inadvertently add a new catch block that accidentally catches the error.

QTT `TestExcutor` uses this pattern and it works well.